drivers/ethernet/eth_mcux.c

```diff
@@ -1669,60 +1688,65 @@ static struct ptp_context ptp_mcux_0_context = {
 	ETH_MCUX_PTP_PINCTRL_INIT(DT_NODELABEL(ptp))
 };
 
-static int ptp_clock_mcux_set(const struct device *dev,
+static int ptp_clock_mcux_get(const struct device *dev,
 			      struct net_ptp_time *tm)
 {
 	struct ptp_context *ptp_context = dev->data;
 	struct eth_context *context = ptp_context->eth_context;
 	enet_ptp_time_t enet_time;
+	uint64_t delta;
+	uint64_t time_ns;
 
-	enet_time.second = tm->second;
-	enet_time.nanosecond = tm->nanosecond;
+	ENET_Ptp1588GetTimerNoIrqDisable(context->base, &context->enet_handle, &enet_time);
 
-	ENET_Ptp1588SetTimer(context->base, &context->enet_handle, &enet_time);
+	k_mutex_lock(&context->ptp_mutex, K_FOREVER);
+	delta = delta_cycles(context, enet_time.nanosecond);
+	context->ptp_cycles += delta;
+	context->last_cycles = enet_time.nanosecond;
+
+	time_ns = context->ptp_cycles + context->offset;
+	ns_to_net_ptp_time(tm, time_ns);
+	k_mutex_unlock(&context->ptp_mutex);
 	return 0;
 }
 
-static int ptp_clock_mcux_get(const struct device *dev,
+static int ptp_clock_mcux_set(const struct device *dev,
 			      struct net_ptp_time *tm)
 {
 	struct ptp_context *ptp_context = dev->data;
 	struct eth_context *context = ptp_context->eth_context;
 	enet_ptp_time_t enet_time;
+	uint64_t req_ns;
+	uint64_t current_ns;
 
-	ENET_Ptp1588GetTimer(context->base, &context->enet_handle, &enet_time);
+	ptp_clock_mcux_get(dev, (struct net_ptp_time *)&enet_time);
+	req_ns = tm->nanosecond + (NSEC_PER_SEC) * tm->second;
+	current_ns = enet_time.nanosecond + (NSEC_PER_SEC * enet_time.second);
 
-	tm->second = enet_time.second;
-	tm->nanosecond = enet_time.nanosecond;
+	k_mutex_lock(&context->ptp_mutex, K_FOREVER);
+	context->offset += req_ns - current_ns;
+        #if ENABLE_HW_TIMER_UPDATE
+	if (context->offset > 10000)
```

Comment:
What is the significance of 10000? Need a comment block or something so that the reader understands why this number.

Reply:
> What is the significance of 10000? Need a comment block or something so that the reader understands why this number.

to avoid adjust the hw timer regularlly, if the gay is too small, use the adjust_ptp_timer would be more stable, other wise there are possbile to get the period jump, which may cause timestamp corrupted. @chencaidy  tell me he find some bugs inside the GPTP, and he will propose a new fix in the stack. I will update our driver accordingly.